### PR TITLE
Add faucet bot filter

### LINKED ISSUE #14: The user request faucet many times in a short time.
## Background / Problem

Too many users are requesting a claim to the faucet at a very frequent rate, and we see some users would spam the /drip command and overload the app.

## Expected Behavior

The same user can only request a token to the bot every hour. This should apply based on their Discord ID so the same person can't request it within 1 hour

### PR DESCRIPTION
**Pull request by y-shijo**
This PR fix #14

### CODE REVIEW

File: src/clients/index.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
 export * from './discord';
 export * from './express';
 export * from './astar';
+export * from './requestFilter';
```
Comment (hoonsubin) [inline src/clients/index.ts:4]: The name `requestFilter`  doesn't look like it should be in the `clients/` folder. What about changing this to something else, or moving the folder and treating it as a database client.

File: src/clients/requestFilter.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import Redis from 'ioredis';
+
+const redis = new Redis(process.env.REDIS_URL);
```
Comment (hoonsubin) [inline src/clients/requestFilter.ts:3]: Please update the `README.md` file and add the environment variable `REDIS_URL`. Also, please add a quick guide on how to get this value.

File: src/clients/requestFilter.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+const anHourInMilliSecond = 60 * 60 * 1000;
+
+// Check whether the requester can request Faucet or not based on the last request time.
+export const canRequestFaucet = async (requesterId: string, now: number): Promise<void> => {
+    const lastRequestAt = Number(await redis.get(requesterId));
+    const untilNextRequest = lastRequestAt + anHourInMilliSecond - now;
```
Comment (hoonsubin) [inline src/clients/requestFilter.ts:9]: I cannot see how long the `untilNextRequest` is supposed to be. I suggest creating a constant that defines how long the cooldown is.
For example:
```ts
// must wait 30 minutes before the next request
const cooldown = 30 * 60 * 1000
```

**Review by hoonsubin — CHANGES_REQUESTED** [review]

**Comment by y-shijo** [comment]
@hoonsubin Please review again :)

File: src/middlewares/requestFilter.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+// Check whether the requester can request Faucet or not based on the last request time.
+export const canRequestFaucet = async (requesterId: string, now: number): Promise<void> => {
+    const lastRequestAt = Number(await redis.get(requesterId));
+    const elapsedTimeFromLastRequest = now - lastRequestAt;
+    
+    // If lastReuqest was made within the cooldown time, the requester cannot request.
+    if (cooldownTimeMillisecond > elapsedTimeFromLastRequest) {
+        const untilNextRequest = cooldownTimeMillisecond - elapsedTimeFromLastRequest;
+        const untilNextRequestMin = Math.floor(untilNextRequest / 1000 / 60);
+        const untilNextRequestSec = Math.floor( (untilNextRequest / 1000) - (untilNextRequestMin * 60) );
```
Comment (hoonsubin) [inline src/middlewares/requestFilter.ts:18]: Please fix the styles. I think everything else looks good!

**Review by hoonsubin — CHANGES_REQUESTED** [review]

**Comment by y-shijo** [comment]
@hoonsubin thank you for your comment. I quickly fixed it, so review it again please 👍

**Comment by hoonsubin** [comment]
@y-shijo the linter and formatter seem to be strange. I'll format the code

**Review by hoonsubin — APPROVED** [review]
LGTM

**Pull request merged**